services/viva/microservice/src/lambdas/syncWorkflow.ts

```diff
@@ -0,0 +1,121 @@
+import log from '../libs/logs';
+import config from '../libs/config';
+import * as dynamoDb from '../libs/dynamoDb';
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import { CaseItem } from '../types/caseItem';
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
+import { VivaWorkflow } from '../types/vivaWorkflow';
+
+type GetCaseResult = Promise<{ Items: CaseItem[] }>;
+interface GetWorkflowParams {
+  personalNumber: string;
+  workflowId: string | null;
+}
+
+export interface GetWorkflowResult {
+  attributes: VivaWorkflow;
+}
+
+export interface Dependencies {
+  getSubmittedOrProcessingOrOngoingCases: (personalNumber: string) => GetCaseResult;
```

Comment:
```js
getSubmittedOrProcessingOrOngoingCases: (personalNumber: string) => Promise<GetCaseResult>;
```

Reply:
Vore fint om det gick att få till detta snyggare där funktionen inte behöver veta som mycket om själva villkoret. Kanske att man skickar in något som funktionen kan lista ut vad som den ska hämta?

Har inte någon konkret lösning men det har säkert vi i teamet?

Reply:
[Typescript har stöd för generics](https://www.typescriptlang.org/docs/handbook/2/generics.html). För att använda dom måste vi in i request-helpern och refaktorera allt som använder den, vilket faller utanför scopet av denna PR

Reply:
> ```js
> getSubmittedOrProcessingOrOngoingCases: (personalNumber: string) => Promise<GetCaseResult>;
> ```

`GetCaseResult` är ett type-alias av `Promise<{ Items: CaseItem[] }>;`